script/passenger-recycler.rb

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env ruby
+#
+# Trivial Passenger memory monitor installed via /etc/cron.hourly/. To modify
+# configuration create /etc/foreman/passenger-recycler.rb.conf file.
+#
+
+CONFIG = '/etc/foreman/passenger-recycler.rb.conf'
+load CONFIG if File.readable?(CONFIG)
+ENABLED ||= true
+MAX_PRIV_RSS_MEMORY ||= 2_000_000
+GRACEFUL_SHUTDOWN_SLEEP ||= 120
+KILL_BUSY ||= true
+SEND_STATUS ||= true
+exit 0 unless ENABLED
+
+def running?(pid)
+  return Process.getpgid(pid) != -1
+rescue Errno::ESRCH
+  return false
+end
+
+require 'phusion_passenger'
+require 'phusion_passenger/platform_info'
```

Comment:
This fails as some environment isn't set up correctly, so the full path to the Passenger installation is unknown:

<pre>
/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- /phusion_passenger/utils/tmpio (LoadError)
        from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /usr/share/passenger/phusion_passenger.rb:235:in `require_passenger_lib'
        from /usr/share/passenger/phusion_passenger/platform_info.rb:26:in `<top (required)>'
        from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require'
        from passenger-recycler.rb:23:in `<main>'
</pre>

with Passenger 5.0.26. Possibly because `PhusionPassenger.locate_directories` isn't called during initialisation.

Should it also use `require_passenger_lib`?